particle-iot/device-os
# photon UDP support

### Problem

UDP Cloud Communications on the Photon/P1. This was not working correctly after refactoring mbedtls in the system firmware and the Gen2/Gen3 integration.

### Solution

- Allow the standard mbedtls config to be used (not the special photon one) when UDP is enabled.
- Weaken the duplicated mbedtls functions from the BCEL library in WICED so that we don't get duplicate 

### Steps to Test
- Edit `hal_platform_compat.h` to enable `HAL_PLATFORM_CLOUD_UDP` on the Photon/P1. 
- Flash monolithic firmware (It does not build with separate system parts.)
- Power on the photon, verify that it connects to the cloud. 
- Verify the handshake is visible via fast blinking cyan LED
- Hit reset
- Verify the handshake is no longer visible and that the device almost immediately connects to the cloud. This is a symptom of UDP session resumption. 
- NB: I did not test that TCP communications still works. UDP is enabled by default with this build. (Verify!)
- Verify WPA Enterprise authentication still works after this change.

### Example App

```c
void setup() {
  /* A minimal example app is super helpful 
   * for testing new features and fixes. 
   * A link to a Docs PR is even better!
   */
}

void loop() {

}
```

### References

Links to the Community, Docs, Other Issues, etc..

---

### Completeness
